content/blog/open-source-monitoring-at-the-hpe-customer-technology-center-böblingen.md

```diff
@@ -0,0 +1,377 @@
+  
+  labels:
+    app: ilo1
+spec:
+  selector:
+    matchLabels:
+      app: ilo1
+  template:
+    metadata:
+      labels:
+        app: ilo1
+    spec:
+      containers:
+        - name: ilo1
+          image: tb1378/iloprometheus:v4.3.1
+          command: ["/usr/bin/python3"]
+          args: ["/opt/prometheus/iloPromConnector.v4.3.py"]
+          volumeMounts:
+            - name: iloprometheusxml
+              mountPath: /opt/prometheus/data
+      volumes:
+        - name: iloprometheusxml
+          configMap:
+            name: iloprom1-xml   # the correct name of the configmap needs to be added here. 
+EOF
+```
+
+After checking that the pod is up and running, I still needed to define a Kubernetes service using this pod/app in order to have at the end the exporter data available for Prometheus.
+
+```shell
+cat << 'EOF' | kubectl apply -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: ilo1
+  namespace: svtprometheus
+#  labels: 
+#     hpecp.hpe.com/hpecp-internal-gateway: "true"    
+spec:
+  selector:
+    app: ilo1
+  ports:
+    - port: 9091               # The Port of that the Prometheus connector uses
+      targetPort: 9091
+      protocol: TCP
+  type: NodePort               # expose the Prometheus connector if you want/need to debug it. 
+EOF
+```
+
+Next, I updated the Prometheus config map to make sure, that the server exporter data is collected and stored in the Prometheus time series database. Hence, I added a job for the server exporter to the existing job list of the Prometheus config map:
+
+```shell
+    - job_name: 'iloredfish'
+      static_configs:
+      - targets: ['ilo1:9091','ilo2:9091','ilo3:9091','ilo4:9091','ilo5:9091','ilo6:9091']
+      honor_timestamps: true
+      scrape_interval: 60s
+      scrape_timeout: 20s
+      metrics_path: /metrics
+      scheme: http
+```
+
+As you can see, I actually use six server exporter (ilo1 up to ilo6) in the CTC monitoring environment to cover the complete list of server hosted in the CTC. Having the data collected into the Prometheus timeseries database I defined some Grafana dashboards to visualize the collected data. I don't want to get here into the details of defining Grafana dashboards - this is well documented in the Grafana documentation. I just want to give you an example of a dashboard I use:
+
+![](/img/ctcracktemppower.2.png "CTC Rack Temperature & Power Overview")
+
+# C﻿onclusion
+
+I tried to show my approach on solving a challenge we had in the Customer Technology Center Böblingen by using available API interfaces and building on an existing Kubernetes, Prometheus and Grafana environment. I decided to write this rather long blog in order to give you all necessary steps I did in detail with the goal to show you what can be done with public APIs, a little bit of Python scripting and the use of open source tools like Prometheus and Grafana. I do hope, that this is helpful when you need to develop your own data collectors.
```

Comment:
Modify the first sentence to:
"In this blog post, I documented my approach on solving a challenge we had in the..."

Remove the comma in the last sentence. Not necessary.

You might want to add some sort of call to action (CTA) to the post. Maybe something like "Check out what the CTC in Boeblingen looks like here." And hotlink to: https://www.hpe.com/us/en/about/customer-centers/boeblingen.html

Just an idea.